File: rename.py
import os

def rename_all(root_dir, old_str, new_str):
    print(f"Renaming {old_str} to {new_str}...")
    
    # 1. Rename contents in all files
    for dirpath, dirnames, filenames in os.walk(root_dir):
        if '.git' in dirpath or 'target' in dirpath:
            continue
        for file in filenames:
            if file == 'rename.py': continue
            filepath = os.path.join(dirpath, file)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                if old_str in content or old_str.capitalize() in content or old_str.upper() in content:
                    content = content.replace(old_str, new_str)
                    content = content.replace(old_str.capitalize(), new_str.capitalize())
                    content = content.replace(old_str.upper(), new_str.upper())
                    
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(content)
                    print(f"Updated content in {filepath}")
            except Exception as e:
                pass # Skip binary files

    # 2. Collect renames bottom up
    renames = []
    for dirpath, dirnames, filenames in os.walk(root_dir, topdown=False):
        if '.git' in dirpath or 'target' in dirpath:
            continue
            
        for name in filenames:
            if old_str in name:
                renames.append((os.path.join(dirpath, name), os.path.join(dirpath, name.replace(old_str, new_str))))
                
        for name in dirnames:
            if old_str in name:
                renames.append((os.path.join(dirpath, name), os.path.join(dirpath, name.replace(old_str, new_str))))

    # 3. Apply renames
    for old, new in renames:
        print(f"Renaming {old} -> {new}")
        os.rename(old, new)
        
    print("Done!")

if __name__ == '__main__':
    rename_all('.', 'yalc', 'ferrox')

File: rename.js
const fs = require('fs');
const path = require('path');

const OLD_NAME = 'yalc';
const NEW_NAME = 'ferrox';
const IGNORE_DIRS = ['.git', 'target', 'node_modules'];

function renameContent(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        if (IGNORE_DIRS.includes(file) || file === 'rename.js' || file === 'rename.py') continue;
        
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        
        if (stat.isDirectory()) {
            renameContent(fullPath);
        } else {
            try {
                let content = fs.readFileSync(fullPath, 'utf8');
                let changed = false;
                
                // Replace variations
                const variations = [
                    [OLD_NAME, NEW_NAME],
                    [OLD_NAME.toUpperCase(), NEW_NAME.toUpperCase()],
                    [OLD_NAME.charAt(0).toUpperCase() + OLD_NAME.slice(1), NEW_NAME.charAt(0).toUpperCase() + NEW_NAME.slice(1)]
                ];
                
                for (const [o, n] of variations) {
                    if (content.includes(o)) {
                        content = content.split(o).join(n);
                        changed = true;
                    }
                }
                
                if (changed) {
                    fs.writeFileSync(fullPath, content, 'utf8');
                    console.log(`Updated content: ${fullPath}`);
                }
            } catch (e) {
                // Ignore binary files or read errors
            }
        }
    }
}

function renameFilesAndDirs(dir) {
    const items = fs.readdirSync(dir);
    
    // Process subdirectories first (bottom-up)
    for (const item of items) {
        if (IGNORE_DIRS.includes(item)) continue;
        const fullPath = path.join(dir, item);
        if (fs.statSync(fullPath).isDirectory()) {
            renameFilesAndDirs(fullPath);
        }
    }
    
    // Now rename items in current dir
    for (const item of items) {
        if (IGNORE_DIRS.includes(item)) continue;
        
        if (item.toLowerCase().includes(OLD_NAME)) {
            const oldPath = path.join(dir, item);
            
            // Replace maintaining case where possible (simplified for just the exact match)
            const newItem = item.split(OLD_NAME).join(NEW_NAME)
                                .split(OLD_NAME.toUpperCase()).join(NEW_NAME.toUpperCase())
                                .split(OLD_NAME.charAt(0).toUpperCase() + OLD_NAME.slice(1)).join(NEW_NAME.charAt(0).toUpperCase() + NEW_NAME.slice(1));
            
            const newPath = path.join(dir, newItem);
            console.log(`Renaming: ${oldPath} -> ${newPath}`);
            fs.renameSync(oldPath, newPath);
        }
    }
}

console.log(`Starting rename from ${OLD_NAME} to ${NEW_NAME}...`);
renameContent('.');
renameFilesAndDirs('.');
console.log('Done!');
